\documentclass{beamer}
\useoutertheme{infolines}
\usepackage[utf8]{inputenc}
%\usepackage{enumitem}
\usepackage{graphicx}
\usepackage[normalem]{ulem}
\newcommand\redout{\bgroup\markoverwith
{\textcolor{red}{\rule[.5ex]{2pt}{2pt}}}\ULon}
\usepackage{xcolor}
\usepackage{textpos}
\usepackage{tikz}   
\usepackage[round]{natbib}
\bibliographystyle{apalike}
\usepackage[T1]{fontenc}
\usepackage{pdfpages}
\usepackage{hyperref}
\usefonttheme[onlymath]{serif}

%\addtobeamertemplate{frametitle}{}{%
%\begin{textblock*}{100mm}(.8\textwidth,-1cm)
%\includegraphics[height=1cm,width=2cm]{opentreelogogrey}
%\end{textblock*}}

\useinnertheme{rectangles}
\usetheme{Copenhagen}
\setbeamertemplate{footline}[frame number]{}
\setbeamertemplate{footline}{}

\title[*]{Phylogenetic likelihood and models}
\author[*]{Emily Jane McTavish}
\institute[*]{
Life and Environmental Sciences\\
University of California, Merced\\
\texttt{ejmctavish@ucmerced.edu, twitter:snacktavish}\\
}
\date{}
\setbeamertemplate{itemize items}[triangle]

\begin{document}

\begin{frame}
\titlepage
(With thanks to Mark Holder and Paul Lewis for slides) 
\end{frame}



\setbeamercolor{background canvas}{bg=}
\includepdf[pages=35]{../../../Lewis/Likelihood2017.pdf}

\setbeamercolor{background canvas}{bg=}
\includepdf[pages=7-8]{../../../markphylo/parsSummaryModels.pdf}


\setbeamercolor{background canvas}{bg=}
\includepdf[pages=36-40]{../../../Lewis/Likelihood2017.pdf}

\setbeamercolor{background canvas}{bg=}
\includepdf[pages=9]{../../../markphylo/parsSummaryModels.pdf}

\begin{frame}
Poisson distribution can be used to explain statistical regularities of rare events\\
$P\left( \textrm{k events in interval} \right) = \frac{{e^{ - \nu } \nu ^k }}{{k!}}$
\begin{itemize}
 \item $\nu$ is the average number of events per interval (rate times time)
 \item $e$ is the number 2.71828... (Euler's number) the base of the natural logarithms
 \item $k$ takes values 0, 1, 2, ...
 \item $k! = k * \left(k - 1\right) * \left(k - 2\right) * … * 2 * 1$ is the factorial of k.
\end{itemize}
\tiny{from wikipedia}
\end{frame}

\begin{frame}
$P\left( \textrm{k events in interval} \right) = \frac{{e^{ - \nu } \nu ^k }}{{k!}}$

$P\left( \textrm{0 events} \right) = \frac{{e^{ - \nu } \nu ^0 }}{{0!}} = e^ {- \nu} = e^{- \mu t}$

$P\left( \textrm{>= 1 events} \right) = 1- e^{- \mu t}$

\end{frame}


\setbeamercolor{background canvas}{bg=}
\includepdf[pages=41-45]{../../../Lewis/Likelihood2017.pdf}

\begin{frame}
\Large
Models for nucleotide substitutions
\end{frame}

\setbeamercolor{background canvas}{bg=}
\includepdf[page=52]{../../../Lewis/Likelihood2017.pdf}

\begin{frame}
Bring in some biology!\\
\includegraphics[height=0.8\textheight]{transition_transversion}\\
Transitions are between two ring purines (A G)
or between one ring pyrimidines (C T).
they therefore involve bases of similar shape. 
Transversions are interchanges of purine for pyrimidine bases.
\end{frame}

\setbeamercolor{background canvas}{bg=}
\includepdf[page=53]{../../../Lewis/Likelihood2017.pdf}

\setbeamercolor{background canvas}{bg=}
\includepdf[page=55-56]{../../../Lewis/Likelihood2017.pdf}

\setbeamercolor{background canvas}{bg=}
\includepdf[pages=59-75]{../../../Lewis/Likelihood2017.pdf}


\begin{frame}
Models matter!
\includegraphics[height=0.6\textheight]{rooting_animal}
\tiny{Yuanning Li, Xing-Xing Shen, Benjamin Evans, Casey W Dunn, Antonis Rokas, Rooting the Animal Tree of Life, Molecular Biology and Evolution, Volume 38, Issue 10, October 2021, Pages 4322–4333, https://doi.org/10.1093/molbev/msab170
}
\end{frame}

\begin{frame}
\Large
How would a model for Copy Number Variant data differ from nucleotide models?\\
What are some components that would make biological sense in a model for CVNs?
\end{frame}






\end{document}
